pytorch_lightning/core/lightning.py

```diff
@@ -1698,22 +1698,34 @@ def get_tqdm_dict(self) -> Dict[str, Union[int, str]]:
                        " and this method will be removed in v1.0.0", DeprecationWarning)
         return self.get_progress_bar_dict()
 
-    def auto_collect_arguments(self):
-        """Collect all arguments module arguments."""
+    def auto_collect_arguments(self) -> None:
+        """
+        Collect all module arguments in the current constructor and all child constructors.
+        The child constructors are all the ``__init__`` methods that reach the current class through
+        (chained) ``super().__init__()`` calls.
+        """
         frame = inspect.currentframe()
 
         frame_args = _collect_init_args(frame.f_back, [])
-        child = _get_latest_child(frame)
```

Comment:
Wasn't there a reason with inheritance for this?

Reply:
I tested it and I found that child is always the specific type, i.e. "self", even if self.auto_collect() is called in a super class. 
So in a concrete example:
```python
class A(object):
    def __init__(self, *args, **kwargs):
        print(self)


class B(A):
    def __init__(self, *args, **kwargs):
        super().__init__()
        print(self)¨

B()  # prints twice <__main__.B object at 0x0000021540C94288>
```
so really it does not matter where we collect "self", it will always be the child (leaf node) in the inheritance tree.
The previous inheritance tests still pass.

Reply:
The `child = _get_latest_child(frame)` does not break anything, it's just redudant as far as I can tell.